From b289de9deb325eeef7919a1f4e3ac1f93b099c16 Mon Sep 17 00:00:00 2001
From: Prabhakar pujeri <prabhakar.pujeri@gmail.com>
Date: Fri, 23 Oct 2020 17:45:16 +0200
Subject: [PATCH] Add new processor characteristics bits from SMBIOS spec 3.4.0

* The bit to indicate 128-bit capable processor
* The bit to indicate processor supports returning ARM64  SoC ID

Signed-off-by: Coiby Xu <coxu@redhat.com>
---
 dmidecode.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dmidecode.c b/dmidecode.c
index 31a49e4..67f6048 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -1388,7 +1388,9 @@ static void dmi_processor_characteristics(const char *attr, u16 code)
 		"Hardware Thread",
 		"Execute Protection",
 		"Enhanced Virtualization",
-		"Power/Performance Control" /* 7 */
+		"Power/Performance Control",
+		"128-bit Capable"
+		"Arm64 SoC ID" /* 9 */
 	};
 
 	if ((code & 0x00FC) == 0)
@@ -1398,7 +1400,7 @@ static void dmi_processor_characteristics(const char *attr, u16 code)
 		int i;
 
 		pr_list_start(attr, NULL);
-		for (i = 2; i <= 7; i++)
+		for (i = 2; i <= 9; i++)
 			if (code & (1 << i))
 				pr_list_item("%s", characteristics[i - 2]);
 		pr_list_end();
-- 
2.31.1

